--- a/wpa_supplicant/config.c
+++ b/wpa_supplicant/config.c
@@ -3335,7 +3335,9 @@
 		if (ret < 0) {
 			if (line) {
 				wpa_printf(MSG_ERROR, "Line %d: failed to "
-					   "parse %s '%s'.", line, var, value);
+					   "parse %s '%s'.", line, field->name,
+					   wpa_debug_show_keys ? value :
+					   "[VALUE REMOVED]");
 			}
 			ret = -1;
 		}
--- a/wpa_supplicant/config_file.c
+++ b/wpa_supplicant/config_file.c
@@ -86,7 +86,8 @@
 {
 	struct wpa_ssid *ssid;
 	int errors = 0, end = 0;
-	char buf[2000], *pos, *pos2;
+	/* Includes 4096 four-byte UTF-8 characters encoded as hexadecimal. */
+	char buf[64 * 1024], *pos, *pos2;
 
 	wpa_printf(MSG_MSGDUMP, "Line: %d - start of a new network block",
 		   *line);
@@ -107,7 +108,8 @@
 		pos2 = os_strchr(pos, '=');
 		if (pos2 == NULL) {
 			wpa_printf(MSG_ERROR, "Line %d: Invalid SSID line "
-				   "'%s'.", *line, pos);
+				   "'%s'.", *line,
+				   wpa_debug_show_keys ? pos : "[VALUE REMOVED]");
 			errors++;
 			continue;
 		}
@@ -116,7 +118,9 @@
 		if (*pos2 == '"') {
 			if (os_strchr(pos2 + 1, '"') == NULL) {
 				wpa_printf(MSG_ERROR, "Line %d: invalid "
-					   "quotation '%s'.", *line, pos2);
+					   "quotation '%s'.", *line,
+					   wpa_debug_show_keys ? pos2 :
+					   "[VALUE REMOVED]");
 				errors++;
 				continue;
 			}
